### DISCUSSION TITLE
cross-referencing: number and optionally title

### DISCUSSION THREAD

**Discussion by espinielli** [quarto-dev/quarto-cli#3583] [Q&A]
I have been searching for the possibility to optionally produce a cross-reference including not only the relevant number but also the title/caption of the chapter/section/table/...
I guess it would be an extension

The idea is to have

```
# My beautiful chapter {#sec-beauty}

# Some other nice chapter {#sec-nice}

See +@sec-beauty
```

which will produce something like 

> See Chapter 1 My beautiful chapter
where "Chapter 1. My beautiful chapter" is a link

Bonus: configuration option to have something like "Chapter 1. (My beautiful chapter)"

Maybe somebody has ideas on how to achieve this (HTML and PDF)?

**Comment by espinielli** [comment]
The current workaround is 
```
See @sec-beauty [My beautiful chapter](#sec-beauty)
```
which will not scale scale well once chapter/section titles are changed...and will need to be tracked in all places where the link is explicitly mentioning them.

**Reply by maucejo** [reply]
Actually, I have the opposition problem. In my Quarto book (v 1.4) for which initial files have been generated using vscode, when I type 
```
# My chapter {#sec-mychapter}

See @sec-mychapter
```
it renders `See 1 My chapter` instead of `See Chapter 1`.

Do you know how to obtint the behavior documented in the Quarto website ?

Thanks

**Reply by cscheid** [reply]
We need more information to be able to help, including the precise version of quarto and a full reproducible example, not just a snippet.

**Reply by maucejo** [reply]
@cscheid Thank you for trying to help me.

Here is the information related to my quarto version
```
quarto check
Quarto 1.4.553
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.553
      Path: C:\ProgramData\chocolatey\lib\quarto\tools\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: Installation From Path
      Path: C:\texlive\2022\bin\win32
      Version: 2022

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.7 (Conda)
      Path: C:/Users/maucejo/anaconda3/python.exe
      Jupyter: 5.5.0
      Kernels: julia-1.10, python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

Regarding the fully reproducible example, I have generated a Book template using `vscode`. The `_quarto.yml` is as follows (I have removed unnecessary files or items) :
```
project:
  type: book

book:
  title: "test_quarto"
  author: "Norah Jones"
  date: "21/04/2024"
  chapters:
    - index.qmd
    - intro.qmd

format:
  html:
    theme: cosmo
```

Finally, for demonstrating my problem with cross-referencing, I modified `intro.qmd` as follows:
```
# Introduction {#sec-intro}

This is a book created from markdown and executable code.

See @knuth84 for additional discussion of literate programming.
```
and referenced this chapter in `index.qmd` as follows:
```
# Preface {.unnumbered}

This is a Quarto book. @sec-intro

To learn more about Quarto books visit <https://quarto.org/docs/books>.
```

The final result is:
![image](https://github.com/quarto-dev/quarto-cli/assets/79322171/bf3a0302-2b42-454b-a60b-6deeeb941938)

As you can see in the screenshot the first chapter is referenced as `1 Introduction` instead of `Chapter 1`.

I hope this description is enough to allow you to reproduce the observed behavior.

Thank you :)

**Reply by cscheid** [reply]
Thank you - I can reproduce the issue.

**Reply by mcanouil** [reply]
FYI:
- https://github.com/quarto-dev/quarto-cli/issues/9439

**Reply by cscheid** [reply]
@maucejo It turns out we document that references within unnumbered sections are unsupported: https://quarto.org/docs/books/book-structure.html#chapter-numbers. We'll improve the situation in the future but no firm timeline as to when this will happen.

**Reply by maucejo** [reply]
@cscheid @mcanouil Thank you for your help. Indeed, the doc is clear. For the moment, I will create the link manually. It is not so difficult 😀
